solarforecastarbiter/reference_forecasts/tests/test_main.py

```diff
@@ -534,3 +537,375 @@ def test_make_latest_nwp_forecasts(forecast_list, mocker, issue_buffer, empty):
+            False
+        ),
+        (
+            forecasts[0], observations[0],
+            pd.Timestamp('2020-05-20T16:00Z'),
+            False
+        )
+    ]
+
+
+def test_generate_reference_persistence_forecast_parameters_no_lead(
+        mocker, perst_fx_obs):
+    forecasts, observations = perst_fx_obs
+    forecasts = [fx.replace(
+        lead_time_to_start=pd.Timedelta('0h'))
+                 for fx in forecasts]
+    session = mocker.MagicMock()
+    session.get_user_info.return_value = {'organization': ''}
+    session.get_observation_time_range.return_value = (
+        pd.Timestamp('2019-01-01T12:00Z'), pd.Timestamp('2020-05-20T16:00Z'))
+    session.get_forecast_time_range.return_value = (
+        pd.Timestamp('2019-01-01T12:00Z'), pd.Timestamp('2020-05-20T14:00Z'))
+    max_run_time = pd.Timestamp('2020-05-20T16:00Z')
+    param_gen = main.generate_reference_persistence_forecast_parameters(
+        session, forecasts, observations, max_run_time
+    )
+    assert isinstance(param_gen, types.GeneratorType)
+    param_list = list(param_gen)
+    assert len(param_list) == 2
+    assert param_list == [
+        (
+            forecasts[0], observations[0],
+            pd.Timestamp('2020-05-20T15:00Z'),
+            False
+        ),
+        (
+            forecasts[0], observations[0],
+            pd.Timestamp('2020-05-20T16:00Z'),
+            False
+        )
+    ]
+
+
+def test_generate_reference_persistence_forecast_parameters_off_time(
+        mocker, perst_fx_obs):
+    forecasts, observations = perst_fx_obs
+    session = mocker.MagicMock()
+    session.get_user_info.return_value = {'organization': ''}
+    session.get_observation_time_range.return_value = (
+        pd.Timestamp('2019-01-01T12:00Z'), pd.Timestamp('2020-05-20T15:33Z'))
+    session.get_forecast_time_range.return_value = (
+        pd.Timestamp('2019-01-01T12:00Z'), pd.Timestamp('2020-05-20T14:10Z'))
+    max_run_time = pd.Timestamp('2020-05-20T16:00Z')
+    # one hour ahead forecast, so 14Z was made at 13Z
+    # enough data to do 14Z and 15Z issue times but not 16Z
+    param_gen = main.generate_reference_persistence_forecast_parameters(
+        session, forecasts, observations, max_run_time
+    )
+    assert isinstance(param_gen, types.GeneratorType)
+    param_list = list(param_gen)
+    assert len(param_list) == 2
+    assert param_list == [
+        (
+            forecasts[0], observations[0],
+            pd.Timestamp('2020-05-20T14:00Z'),
+            False
+        ),
+        (
+            forecasts[0], observations[0],
+            pd.Timestamp('2020-05-20T15:00Z'),
+            False
+        ),
+    ]
+
+
+def test_generate_reference_persistence_forecast_parameters_multiple(
+        mocker, perst_fx_obs):
+    forecasts, observations = perst_fx_obs
+    forecasts[0] = forecasts[0].replace(
+        extra_parameters=(forecasts[0].extra_parameters[:-1] +
+                          ', "index_persistence": true}')
```

Comment:
Do we need a test that asserts that a persistence index function is called when `"index_persistence": true`?

Reply:
we'll leave that up to the ``run_persistence`` function and just make sure the value of index is correct when it is called